proxy_agent/src/service.rs

```diff
@@ -4,20 +4,18 @@
 pub mod windows;
 
 use crate::common::{config, constants, helpers, logger};
-use crate::proxy::proxy_listener;
-use crate::shared_state::{telemetry_wrapper, SharedState};
+use crate::proxy::proxy_server;
+use crate::shared_state::{shared_state_wrapper, telemetry_wrapper, SharedState};
 use crate::telemetry::event_reader;
 use proxy_agent_shared::logger_manager;
 use proxy_agent_shared::telemetry::event_logger;
 use std::sync::{Arc, Mutex};
 use url::Url;
 
-#[cfg(not(windows))]
-use std::thread;
 #[cfg(not(windows))]
 use std::time::Duration;
 
-pub fn start_service(shared_state: Arc<Mutex<SharedState>>) {
+pub async fn start_service(shared_state: Arc<Mutex<SharedState>>) {
```

Comment:
you are right, the async is no longer needed after I moved to tokio::spawn instead of calling the async methods.

reviewed all the async keywords in the project and removed unnecessary ones.